src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.internet.shop.dao.impl;
+
+import java.util.Optional;
+
+import mate.academy.internet.shop.dao.BucketDao;
+import mate.academy.internet.shop.database.Storage;
+import mate.academy.internet.shop.lib.Dao;
+import mate.academy.internet.shop.model.Bucket;
+
+@Dao
+public class BucketDaoImpl implements BucketDao {
+    private static Long bucketCounter = 1L;
+
+    @Override
+    public Bucket create(Bucket bucket) {
+        bucket.setId(bucketCounter);
+        Storage.buckets.add(bucket);
+        bucketCounter++;
+        return bucket;
+    }
+
+    @Override
+    public Bucket update(Bucket bucket) {
+        for (Bucket b : Storage.buckets) {
+            if (b.getId().equals(bucket.getId())) {
+                b.setSelectedItems(bucket.getSelectedItems());
+            }
+        }
+        return bucket;
```

Comment:
try to use Stream api

Reply:
You mean?         return Storage.buckets
                .stream()
                .filter(b -> b.getId().equals(bucket.getId()))
                .findFirst()
                .ifPresent(b -> b.setSelectedItems(bucket.getSelectedItems());

Reply:
Or ?         Storage.buckets
                .stream()
                .filter(b -> b.getId().equals(bucket.getId()))
                .findFirst()
                .ifPresent(b -> b.setSelectedItems(bucket.getSelectedItems()));
        return bucket;

Reply:
Done elsewhere